examples/DotNetCore/WebDemoWithEventHub/WebDemoWithEventHub/EventHubService.cs

```diff
@@ -0,0 +1,86 @@
+﻿using Azure.Identity;
+using Azure.Messaging.EventHubs;
+using Azure.Messaging.EventHubs.Consumer;
+using Azure.Messaging.EventHubs.Processor;
+using Azure.Storage.Blobs;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Configuration.AzureAppConfiguration;
+using Microsoft.Extensions.Options;
+using System;
+using System.Threading.Tasks;
+
+namespace WebDemoWithEventHub
+{
+    public class EventHubService
+    {
+        private Settings _settings;
+
+        private EventHubConnection _eventHubConnection;
+
+        private EventProcessorClient _processorClient;
+
+        private IConfigRefresher _configRefresher;
+
+        public EventHubService(IConfiguration config, IConfigRefresher configRefresher)
+        {
+            _configRefresher = configRefresher;
+
+            _eventHubConnection = config.GetSection("WebDemo:Connection").Get<EventHubConnection>();
+            _settings = config.GetSection("WebDemo:Settings").Get<Settings>();
+
+            InitEventHubProcessor();
+        }
+
+        private void InitEventHubProcessor()
```

Comment:
The `_eventHubConnection` is only used in this function. We can make it a parameter passed this function.

Reply:
Changed it to an arg.

---
In reply to: [427597726](https://github.com/Azure/AppConfiguration/pull/329#discussion_r427597726) [](ancestors = 427597726)